nuimo.py

```diff
@@ -1,376 +1,439 @@
+    :ivar gesture: gesture that was performed
+    :ivar value: value associated with the gesture, i.e. number of rotation steps
+    """
+    def __init__(self, gesture, value=None):
+        self.gesture = gesture
+        self.value = value
 
-    def start_discovery(self):
-        self.log("started Nuimo discovery")
-        if self.delegate:
-            self.delegate.discovery_started()
-        all_discovered_devices = self.discovery_service.discover(int(self.timeout))
-        self.log_devices(all_discovered_devices)
-        self.nuimos = self.create_nuimos(all_discovered_devices)
-        self.log_nuimos(self.nuimos)
-        self.fire_callbacks()
-        if self.delegate:
-            self.delegate.discovery_finished()
-
-    def stop_discovery(self):
-        self.log("stopped Nuimo discovery")
-
-    @staticmethod
-    def filter_nuimos(devices):
-        return [
-            addr for addr, attrs in devices.items()
-            if attrs and attrs.get('name') == 'Nuimo'
-        ]
+    def __repr__(self):
+        return str(self.gesture) + (("," + str(self.value)) if self.value is not None else "")
 
-    def create_nuimos(self, devices):
-        return [NuimoController(device) for device in self.filter_nuimos(devices)]
 
-    def fire_callbacks(self):
-        if not self.delegate:
-            return
+class Gesture(Enum):
+    """
+    A gesture that can be performed by the user on a Nuimo controller.
+    """
+    BUTTON_PRESS = 1
+    BUTTON_RELEASE = 2
+    SWIPE_LEFT = 3
+    SWIPE_RIGHT = 4
+    SWIPE_UP = 5
+    SWIPE_DOWN = 6
+    TOUCH_LEFT = 8,
+    TOUCH_RIGHT = 9,
+    TOUCH_TOP = 10,
+    TOUCH_BOTTOM = 11,
+    LONGTOUCH_LEFT = 12
+    LONGTOUCH_RIGHT = 13
+    LONGTOUCH_TOP = 14,
+    LONGTOUCH_BOTTOM = 15,
+    ROTATION = 16,
+    FLY_LEFT = 17,
+    FLY_RIGHT = 18,
+    FLY_UPDOWN = 19
+
+
+class LedMatrix:
+    """
+    Represents an LED matrix to be displayed on a Nuimo controller.
 
-        [self.delegate.controller_added(nuimo) for nuimo in self.nuimos]
+    :ivar leds: Boolean array with 81 values each representing the LEDs being on or off.
+    """
+    def __init__(self, string):
+        """
+        Initializes an LED matrix with a string where each character represents one LED.
 
-    def log_devices(self, devices):
-        for device, attrs in devices.items():
-            self.log("discovered Bluetooth device {}, {}".format(device, attrs))
+        :param string: 81 character string: ' ' and '0' represent LED off, all other characters represent LED on.
+        """
+        string = '{:<81}'.format(string[:81])
+        self.leds = [c not in [' ', '0'] for c in string]
 
-    def log_nuimos(self, nuimos):
-        for nuimo in nuimos:
-            self.log(nuimo)
+    def __eq__(self, other):
+        return (other is not None) and (self.leds == other.leds)
 
-    def log(self, msg):
-        _LOGGER.info("%s, %s", self.adapter, msg)
+    def __ne__(self, other):
+        return not self == other
```

Comment:
According to http://stackoverflow.com/questions/1227121/compare-object-instances-for-equality-by-their-attributes-in-python#comment-22555187:

> One caveat: defining __eq__ does not automatically add support for the != operator